### PR TITLE
JavaScript capstone project - collaporative

### PR DESCRIPTION
**Pull request by Elerqsousy**
### Team
The team consists of 2 students only:
- Ayoub Chahir (Student A)
- Mahmoud Rizk (Student B)

### Requirements

- Used JS ES6 syntax and modules
- Used webpack
- Used Jest for testing
- Followed Github flow
- Used the involvement API for comment, likes, and reservation handling
- Used the TV-maze API for main content handling
- Performed peer-to-peer code review for each project task

### Project Features:

1. Upon page load the main page gets populated with 6 shows from the app
2. Each show reflects the number of likes for each show on the main page
3. Upon clicking on the comment button of the app, a modal pops up presenting more details about each app
4. the Modal (show details) has a comment and reservation section
5. Upon Modal load comments and reservations gets loaded from the involvement API
6. Each popup presents the number of comments and reservations for each show.

<br>

For a demo Presentation for the App please use this [link](https://drive.google.com/file/d/1MhHWGp7qNawExu1sP-Odo4PGh5FPm4X9/view?usp=sharing).

### CODE REVIEW

File: src/modules/api-comment/comments-build.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import getcomments from './get-comments.js';
+import commentCount from './comment-count.js';
+
+const commentBuild = async (containerClass, url, id, itemID, commentCountContainer) => {
```
Comment (ShoiraTa) [inline src/modules/api-comment/comments-build.js:4]: - [ ] You have done a great job working on the project, all the functionality works as expected. Props on that 👏🏻 . But I noticed that you have 1 modal for both, comments and reservations. However, based on project requirements, you should be having separate modals for comments and reservations: 
Screenshot 2022-03-11 at 13 59 19

- [ ] Please make sure to add the `reservations` button below the `comment` button on the index page, and render a separate modals:

Wireframe layout   | Your layout
------------- | -------------
Screenshot 2022-03-11 at 14 01 44 | Screenshot 2022-03-11 at 14 01 24

**Review by ShoiraTa — CHANGES_REQUESTED** [review]
Hi team,

:thumbsup: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

There are some issues that you still need to work on to go to the next project but you are almost there!



 ## Required Changes  :recycle: 

_Check the comments under the review._ 

***
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear by mentioning my Github username.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

* * *
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/api-comment/comments-build.js (hunk 1 above)
Comment (Elerqsousy) [reply src/modules/api-comment/comments-build.js:4]: @ShoiraTa Thank you very much for your time and feedback. 
We are a team of 2 only, and I implemented the reservation part as I found it challenging, but it is not part of 2-member team requirements. 

We will be working on modifying the project later on, but for now, the reservation part was extra work I did on the project.

Thank you once again and please, advise if we still need to work on this part.

Comment (ShoiraTa) [reply src/modules/api-comment/comments-build.js:4]: @Elerqsousy  I saw three contributors in this project, it looked like you are a team of 3 😁 sorry for that.  Please request a review again and the project will be approved.

**Review by elmar8287 — APPROVED** [review]
Hi, 

# APPROVED  👍 🥇 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**